### PR TITLE
feat: add @Exclude to suppress auto-copy defaults for all 6 copy annotations

### LINKED ISSUE #8: Support `@CopyProperty.Exclude`
Exclude some properties from being copied and make sure they are specified.

### PR DESCRIPTION
**Pull request by TBSten**
## Summary

Implements [issue #8](https://github.com/TBSten/cream/issues/8) — **make specific properties required in generated copy functions** by stripping their \`= this.<property>\` auto-copy default.

- **Semantics (A)**: \`@Exclude\` keeps the parameter in the signature but removes its default, forcing the caller to supply an explicit value. The parameter is **not** removed.
- **6 annotations covered**: \`@CopyTo\`, \`@CopyFrom\`, \`@CombineTo\`, \`@CombineFrom\`, \`@SealedCopy\`, \`@CopyToChildren\`
- **Annotation-scoped**: \`@SealedCopy.Exclude\` affects only \`copy()\`; \`@CopyToChildren.Exclude\` affects only per-child \`copyToCn()\` — both can coexist on the same sealed type without interference.
- **No-op warning**: \`@Exclude\` on an unmatched parameter emits a KSP \`warn("... has no effect: not a matched property")\`.

<details>
<summary><b>Quick example</b></summary>

\`\`\`kotlin
sealed interface State {
    val name: String
    val count: Int

    @CopyFrom(State::class)
    data class Success(
        val name: String,
        @CopyFrom.Exclude val count: Int,  // no default → required
    )
}

// Generated:
fun State.copyToStateSuccess(
    name: String = this.name,
    count: Int,              // required — caller must specify
): State.Success = ...
\`\`\`

</details>

<details>
<summary><b>Changes overview</b></summary>

**runtime** (\`cream-runtime\`):
- \`CopyTo.Exclude\`, \`CopyFrom.Exclude\` — \`@Target(VALUE_PARAMETER)\`, field-less markers
- \`CombineTo.Exclude\` — \`@Target(PROPERTY)\` (mirrors \`CombineTo.Map\`)
- \`CombineFrom.Exclude\` — \`@Target(VALUE_PARAMETER)\`
- \`SealedCopy.Exclude\`, \`CopyToChildren.Exclude\` — \`@Target(PROPERTY)\` for sealed parent abstract properties
- \`CopyFrom.Exclude\`: replaces the pre-existing dead-code stub \`Exclude(val value: String)\`

**ksp** (\`cream-ksp\`):
- \`ExcludeProperty.kt\` — annotation-scoped \`isExcludedFromCopy()\` + warning helpers
- \`Class.kt\` — gate \`= this.x\` default with \`!isExcludedFromCopy()\`
- \`CombineToClass.kt\` — same; the default is suppressed when the matched property is \`@Exclude\`-marked in **any** contributing source (not only the overlap "winner")
- \`SealedCopy.kt\` — gate \`= this.propName\` in \`appendSealedCopyHeader\`
- \`CreamSymbolProcessor\` + \`CreamSymbolProcessorProvider\` — add \`KSPLogger\`
- Processors — thread \`logger\`; warn for unmatched Excludes (target-side and source-side) and non-abstract sealed properties with \`@Exclude\`

**tests**:
- \`ExcludeSnapshotTest\` — kctfork snapshot tests for all 6 annotations + edge cases (all names in English)
- \`ExcludeTest\` (integration) — happy-path verification that caller-supplied values are reflected

**docs**: \`README.md\` + \`README.ja.md\` — new \`### CopyTo.Exclude, ...\` section after \`.Map\`

</details>

<details>
<summary><b>Exploratory debugging & hardening</b></summary>

The feature was stress-tested with a behavioral probe across its combinatorial surface (annotation × placement × class kind × multi-source/target × `.Map`). Three correctness defects surfaced and are fixed here, each with a regression test:

- **`@CombineTo` multi-target no-op warning fired once per target** (duplicate). Now computed once per source over the union of all its targets' parameters, mirroring `CopyToProcessor`.
- **`@CombineTo` multi-target false-positive warning**: an exclude matched in one target but absent in a sibling target wrongly warned "has no effect". Fixed by the same per-source union.
- **`@CombineTo` exclude silently ignored on a shadowed source**: with overlapping property names across sources, the default is now suppressed when the property is `@Exclude`-marked in **any** contributing source, not only the overlap "winner".
- **Target-side no-op warning suppressed on the sealed-target fan-out** (`@CopyFrom`/`@CombineFrom` to a sealed interface): `logger` is now threaded into the source→subclass generation so the warning fires once.

Verified-correct (no change needed): `@SealedCopy.Exclude` with `RETURN_NULL`/`RETURN_AS_IS` + object subtypes, `.Map` + `.Exclude`, generics, nullable types, nested sealed hierarchies, and `@SealedCopy`/`@CopyToChildren` annotation-scoping.

Also corrected an inaccurate `@CombineFrom.Exclude` KDoc claim ("applies to all generated overloads" — repeated `@CombineFrom` actually merges into a single function), and added a `# Exclude` section to each parent annotation's KDoc.

A follow-up review of the snapshot inputs/outputs from a **user's perspective** surfaced a (pre-existing, cream-wide) bug, fixed in a separate `fix(ksp)` commit: the generated KDoc `Example` blocks hardcoded a no-arg call `source.<fun>()` even when a parameter was required (e.g. via `@Exclude`), so the documented call would not compile. The examples now include the required parameters. A `test:` commit also strengthens coverage (excluded-param-first ordering, `@CopyTo.Map` + `@Exclude`, target-default-value behavior, and a realistic generics shape replacing a degenerate `data class T<T>`).

</details>

## Test plan

- [x] \`./gradlew :cream-ksp:test\` — all existing snapshots unchanged; new \`ExcludeSnapshotTest\` passes
- [x] \`./gradlew :test:jvmTest\` — integration tests green
- [x] \`./gradlew jvmTest ktlintCheck\` — full JVM test suite + lint clean
- [ ] \`./gradlew iosSimulatorArm64Test\` — native coverage (CI)
- [ ] \`./gradlew testDebugUnitTest testReleaseUnitTest\` — Android (CI)

Closes #8

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cream-ksp/src/test/resources/snapshots/ExcludeSnapshotTest/edgeCase/Map and Exclude combined remove default.md
Comment (TBSten) [inline cream-ksp/src/test/resources/snapshots/ExcludeSnapshotTest/edgeCase/Map and Exclude combined remove default.md:1]: snapshot ファイル名は 英語にして... (他も同じ）

File: cream-ksp/src/main/kotlin/me/tbsten/cream/ksp/transform/ExcludeProperty.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            matchedProperty?.isSourcePropertyExcluded(matchedSource, CopyTo.Exclude::class) == true
+        is GenerateSourceAnnotation.CopyFrom ->
+            annotationsOf(CopyFrom.Exclude::class).any()
+        is GenerateSourceAnnotation.CombineTo ->
+            matchedProperty?.isSourcePropertyExcluded(matchedSource, CombineTo.Exclude::class) == true
+        is GenerateSourceAnnotation.CombineFrom ->
+            annotationsOf(CombineFrom.Exclude::class).any()
+        is GenerateSourceAnnotation.CopyToChildren ->
+            matchedProperty?.annotationsOf(CopyToChildren.Exclude::class)?.any() == true
+        else -> false
```
Comment (TBSten) [inline cream-ksp/src/main/kotlin/me/tbsten/cream/ksp/transform/ExcludeProperty.kt]: else は使わないで

File: cream-runtime/src/commonMain/kotlin/me/tbsten/cream/CombineFrom.kt
Comment (TBSten) [inline cream-runtime/src/commonMain/kotlin/me/tbsten/cream/CombineFrom.kt:37]: 元のアノテーションの KDoc に # Exclude セクションを追加して (全アノテーション対応必須)

Comment (TBSten) [reply cream-runtime/src/commonMain/kotlin/me/tbsten/cream/CombineFrom.kt:37]: 対応しました。6 つの親アノテーション (CopyTo / CopyFrom / CombineTo / CombineFrom / SealedCopy / CopyToChildren) のクラスレベル KDoc に `# Exclude` セクションと `@see X.Exclude` を追加しました (feat(runtime) コミット)。あわせて `@CombineFrom.Exclude` の KDoc にあった「generated overloads」という不正確な記述 (実際は 1 関数にマージされる) も修正しています。
